rpcs3/Emu/System.cpp

```diff
@@ -1061,7 +1060,8 @@ game_boot_result Emulator::Load(const std::string& title_id, bool add_only, bool
 
 		if (!fs::is_file(g_cfg.vfs.get_dev_flash() + "sys/external/liblv2.sprx"))
 		{
-			return game_boot_result::firmware_missing;
+			extern void show_firmware_missing_message_box();
+			show_firmware_missing_message_box();
```

Comment:
is this guaranteed to be called from the gui thread ?

Reply:
Good question, in code Emu.Load is called only in main_window::OnPlayOrPause which is connected to some Qt connect() method.
Depends on which thread executes qt connect() callbacks.